### PR TITLE
feat: cache HTTP calls for fetching secret shares

### LINKED ISSUE #126: cache half secret HTTP calls
temporarily cache half secret HTTP calls

### PR DESCRIPTION
**Pull request by aholovko**
closes #126

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>
Signed-off-by: Andriy Holovko <andriy.holovko@gmail.com>

### CODE REVIEW

File: test/bdd/fixtures/kms/docker-compose.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ services:
       - KMS_KEY_MANAGER_STORAGE_PREFIX=authzkmskm
       - KMS_HUB_AUTH_URL=https://hub-auth.example.com:8070
       - KMS_HUB_AUTH_API_TOKEN=test_token
+      - KMS_CACHE_EXPIRATION=10m
```
Comment (DRK3) [inline test/bdd/fixtures/kms/docker-compose.yml:34]: This env variable is missing from your start command - need to add it there and pass it in to your cache provider

**Comment by DRK3** [comment]
`chore:`? I think it should be `feat:`

**Comment by aholovko** [comment]
> `chore:`? I think it should be `feat:`

title comes from the original commit authored by Baha, and I wanted to preserve its original roots - but I agree that it might be improved...

File: test/bdd/fixtures/kms/docker-compose.yml (hunk 1 above)
Comment (aholovko) [reply test/bdd/fixtures/kms/docker-compose.yml:34]: support for that env was implemented in [my previous PR](https://github.com/trustbloc/hub-kms/pull/128) when caching for EDV was added, here I am reusing that functionality for Hub Auth calls as well

**Review by fqutishat — APPROVED** [review]

File: pkg/secretlock/secretsplitlock/secret_split_lock.go [hunk 2]
```diff
@@ -81,37 +120,49 @@ func New(secret []byte, params *HubAuthParams, options ...Option) (secretlock.Se
...
+		return cachedSecret, nil
 	}
+
+	if errors.Is(getErr, ariesstorage.ErrDataNotFound) {
+		secret, fetchErr := fetchFunc()
+		if fetchErr != nil {
+			return nil, fetchErr
+		}
+
+		putErr := cache.Put(cacheKey, secret)
```
Comment (llorllale) [inline pkg/secretlock/secretsplitlock/secret_split_lock.go:145]: This cache seems to just grow. Should have some eviction policy.

Comment (aholovko) [reply pkg/secretlock/secretsplitlock/secret_split_lock.go:145]: for now it's based on expiration time

Comment (llorllale) [reply pkg/secretlock/secretsplitlock/secret_split_lock.go:145]: Why didn't you reuse `bluele/gcache` ?

Comment (llorllale) [reply pkg/secretlock/secretsplitlock/secret_split_lock.go:145]: Nevermind.

**Review by llorllale — APPROVED** [review]

File: pkg/secretlock/secretsplitlock/secret_split_lock.go (hunk 2 above)
Comment (aholovko) [reply pkg/secretlock/secretsplitlock/secret_split_lock.go:145]: `cache` in the snippet above is cacheStore that satisfies our `storage.Store` interface - under the hood it uses `bluele/gcache` - [here](https://github.com/trustbloc/hub-kms/blob/master/pkg/storage/cache/gcache_store.go) is an implementation

File: test/bdd/fixtures/kms/docker-compose.yml (hunk 1 above)
Comment (DRK3) [reply test/bdd/fixtures/kms/docker-compose.yml:34]: Oh, I see - so it in this branch the env key doesn't exists, but in the master branch it does? (So when merged, it'll be there?)

Comment (DRK3) [reply test/bdd/fixtures/kms/docker-compose.yml:34]: My mistake - I had checked out the wrong branch locally (there was one in the upstream that's similar but not quite the right onw)

**Pull request merged**